feat: added moonbeam to human protocol subgraph list

tnkerer opened the pull request.

tnkerer committed:
Initial commit

tnkerer committed:
initial commit

tnkerer committed:
feat: merge moonbeam subgraph source code

tnkerer committed:
fix: fixed merge conflicts

tnkerer committed:
feat: add workflow for moonbeam deployment

tnkerer committed:
docs: added moonbeam subgraph address to docs

tnkerer committed:
fix: rolled back dependencies version to match original

tnkerer committed:
fix: removed lock files

posix4e commented on abis/EscrowFactory.json:
```diff
@@ -30,42 +30,44 @@
     "name": "counter",
     "outputs": [{ "internalType": "uint256", "name": "", "type": "uint256" }],
     "stateMutability": "view",
-    "type": "function"
```
let's not update this file unless we need to

Jalal-1 replied:
Agreed

posix4e commented on package.json:
```diff
@@ -1,16 +1,17 @@
     "@graphprotocol/graph-cli": "0.29.2",
-    "@graphprotocol/graph-ts": "0.26.0"
+    "@graphprotocol/graph-ts": "0.26.0",
+    "gluegun": "^5.1.2"
```
Why do we want gluegun?

posix4e commented on package.json:
```diff
@@ -1,16 +1,17 @@
-    "create-local": "graph create --node http://localhost:8020/ posix4e/humansubgraph",
-    "remove-local": "graph remove --node http://localhost:8020/ posix4e/humansubgraph",
-    "deploy-local": "graph deploy --node http://localhost:8020/ --ipfs http://localhost:5001 posix4e/humansubgraph"
+    "deploy": "graph deploy --node https://api.thegraph.com/deploy/ menezesphill/hmtmoonbeam",
```
i don't think we need to update this stuff. We probably should remove them altogether

posix4e reviewed (changes requested):
Have some changes in the PR

tnkerer committed:
fix: resolving requested changes

tnkerer commented:
> Have some changes in the PR

@posix4e I reverted ABIs back to the original from the main branch. Removed `create-local`, `remove-local`, `deploy-local`, and `deploy` scripts from package.json and removed `gluegun` since it is a fix unrelated to this deployment for an error that I had encountered while using yarn.

ivhus reviewed (dismissed):
So that indicating environment variables in a way like `USER_ID=239482 USER_KEY=foobar node app.js` works

ivhus reviewed (approved):
The deploy part looks fine

Not sure about package.json and such 


There's a question with humanprotocol/moonbeam (https://thegraph.com/hosted-service/subgraph/humanprotocol/moonbeam):

There was no subgraph like this, I am not sure it's going to be created automatically (only the deploy to pre-created subgraph was tested before).

tnkerer commented:
> The deploy part looks fine
> 
> Not sure about package.json and such
> 
> There's a question with humanprotocol/moonbeam (https://thegraph.com/hosted-service/subgraph/humanprotocol/moonbeam):
> 
> There was no subgraph like this, I am not sure it's going to be created automatically (only the deploy to pre-created subgraph was tested before).

Hi @ivhus, I believe package.json is ok since workflow only uses `codegen` & `build`, these two scripts are still there. 

From my experience, the subgraph is not created automatically, so I believe the deployment will only take effect if the humanprotocol/moonbeam subgraph is created in advance using the foundation account.

posix4e commented:
@ivhus i tagged you in the ticket so we could use a foundation controlled subgraph, like the other two production nets

ivhus commented:
https://thegraph.com/hosted-service/subgraph/humanprotocol/moonbeam 

so I assume after the merge it should be deployed

posix4e commented:
You will need to create the project

tnkerer commented:
> https://thegraph.com/hosted-service/subgraph/humanprotocol/moonbeam
> 
> so I assume after the merge it should be deployed

Confirmed, the repo has been created on The Graph. I think we are good to go.
